### PR TITLE
Deprecate `ca_signature_algo` config

### LINKED ISSUE #12905: Remove `ca_signature_algo` configuration
Expected behavior:
After the merge of https://github.com/gravitational/teleport/pull/12674 we no longer use the following configuration:
```yaml
teleport:
    ca_signature_algo: "rsa-sha2-512"
```
As we now rely upon the `x/crypto` package to choose the signing algorithm

Current behavior:

Bug details:
- Teleport version: https://github.com/gravitational/teleport/pull/12674 (after 9.2.4)
- Recreation steps: remove the relevant code and everything should work the same way as before
- Debug logs

### PR DESCRIPTION
**Pull request by marcoandredinis**
After the merge of https://github.com/gravitational/teleport/pull/12674 we no longer use the following configuration:
```yaml
teleport:
    ca_signature_algo: "rsa-sha2-512"
```
As we now rely upon the `x/crypto` package to choose the signing algorithm (it defaults to `rsa-sha2-512`)

**Demo**
If we set `ca_signature_algo` (the value is irrelevant) and start `teleport` we get:
```shell
root@marco:/workspace# teleport start --debug
2022-06-02T09:33:58Z WARN             ca_signing_algo config option is deprecated and will be ignored, we'll always default to rsa-sha2-512. config/configuration.go:348
2022-06-02T09:33:58Z INFO             Generating new host UUID: b001159a-10e0-49a7-b4dc-61c73fbe9e42. service/service.go:726
...
```

Fixes #12905

### CODE REVIEW

File: lib/config/fileconf.go [hunk 1]
```diff
@@ -420,9 +414,6 @@ func (conf *FileConfig) CheckAndSetDefaults() error {
 			return trace.BadParameter("MAC algorithm %q is not supported; supported algorithms: %q", m, sc.MACs)
 		}
 	}
-	if conf.CASignatureAlgorithm != nil && !apiutils.SliceContainsStr(validCASigAlgos, *conf.CASignatureAlgorithm) {
```
Comment (codingllama) [inline lib/config/fileconf.go:423]: Let's add a deprecation notice at fileconf.go / Global.CASignatureAlgorithm too, so it's clear the field is unused and kept only for config backwards compatibility.

Is there a "Cloud"/protobuf counterpart for this setting, or just the fileconf / teleport.yaml one?

**Review by codingllama — COMMENTED** [review]
Thanks for the cleanup, @marcoandredinis !

File: lib/config/fileconf.go (hunk 1 above)
Comment (marcoandredinis) [reply lib/config/fileconf.go:423]: Deprecation notice added :pray: 

> Is there a "Cloud"/protobuf counterpart for this setting, or just the fileconf / teleport.yaml one?

CertAuthorityV2 has two methods that I'm pretty sure could be removed: `Get/SetSigningAlg` 

My concern is having a version without those methods interacting with a version that has them
I don't know how that works out (backwards/forward compat)
I'll do some tests :hammer_and_wrench:

Comment (codingllama) [reply lib/config/fileconf.go:423]: If you drop the methods and an older gRPC client tries to call them, it'll get a framework-level error saying the method doesn't exist (I don't remember the exact error, but it's something like that).

Whether it's safe to drop depends on how it's used. If you can ensure all calls are removed by version N, then it's safe to drop the method in N+1.

**Comment by codingllama** [comment]
LGTM Marco, let me know if you intend to do further work in this PR, otherwise I'm happy to approve.

**Review by zmb3 — APPROVED** [review]
Looks good, but there's also a reference to `ca_signature_algo` in our docs - please update that as well.

**Comment by marcoandredinis** [comment]
> LGTM Marco, let me know if you intend to do further work in this PR, otherwise I'm happy to approve.

@codingllama Let's go forward with this one

**Comment by marcoandredinis** [comment]
@zmb3 @codingllama 
Can you please take another look?
I removed some more files, and I'm confident the only leftovers are the gRPC messages, which can be removed in the next major version

I tested using teleport built from this branch against `tsh` v9.3.2 and it worked:
```shell
root@moon:/workspace/teleport# tsh version
Teleport v10.0.0-dev git:teleport-connect-preview-1.0.0.dev.1-252-g855e461ca go1.17.10
Proxy version: 10.0.0-dev
root@moon:/workspace/teleport# tsh ssh root@marco.mydemo
root@marco:~# exit
logout
the connection was closed on the remote side on  03 Jun 22 10:45 UTC
root@moon:/workspace/teleport# ./tsh version
Teleport v9.3.2 git:v9.3.2-0-gb3597d12a go1.17.10
root@moon:/workspace/teleport# ./tsh ssh root@marco.mydemo
root@marco:~# exit
logout
the connection was closed on the remote side on  03 Jun 22 10:45 UTC
```

File: integration/helpers.go [hunk 2]
```diff
@@ -297,8 +296,7 @@ func (s *InstanceSecrets) GetCAs() ([]types.CertAuthority, error) {
 				Cert:    s.TLSCACert,
 			}},
 		},
-		Roles:      []string{},
-		SigningAlg: types.CertAuthoritySpecV2_RSA_SHA2_512,
+		Roles: []string{},
```
Comment (codingllama) [inline integration/helpers.go:299]: nit: drop this too?

```suggestion
```

File: integration/helpers.go [hunk 3]
```diff
@@ -336,8 +333,7 @@ func (s *InstanceSecrets) GetCAs() ([]types.CertAuthority, error) {
...
```
Comment (codingllama) [inline integration/helpers.go:336]:
```suggestion
```

File: lib/auth/clt_test.go [hunk 4]
```diff
@@ -162,8 +162,7 @@ func newCertAuthority(t *testing.T, name string, caType types.CertAuthType) type
 				Key:  key,
 			}},
 		},
-		Roles:      nil,
-		SigningAlg: types.CertAuthoritySpecV2_RSA_SHA2_256,
+		Roles: nil,
```
Comment (codingllama) [inline lib/auth/clt_test.go:165]:
```suggestion
```

File: lib/config/configuration.go [hunk 5]
```diff
@@ -1493,8 +1493,7 @@ func parseAuthorizedKeys(bytes []byte, allowedLogins []string) (types.CertAuthor
 				PublicKey: ssh.MarshalAuthorizedKey(pubkey),
 			}},
 		},
-		Roles:      nil,
-		SigningAlg: types.CertAuthoritySpecV2_UNKNOWN,
+		Roles: nil,
```
Comment (codingllama) [inline lib/config/configuration.go:1496]:
```suggestion
```

File: lib/reversetunnel/srv_test.go [hunk 6]
```diff
@@ -53,8 +52,7 @@ func TestServerKeyAuth(t *testing.T) {
 				PublicKey:      pub,
 			}},
 		},
-		Roles:      nil,
-		SigningAlg: types.CertAuthoritySpecV2_RSA_SHA2_256,
+		Roles: nil,
```
Comment (codingllama) [inline lib/reversetunnel/srv_test.go:55]:
```suggestion
```

File: lib/services/watcher_test.go [hunk 7]
```diff
@@ -866,8 +866,7 @@ func newCertAuthority(t *testing.T, name string, caType types.CertAuthType) type
 				},
 			},
 		},
-		Roles:      nil,
-		SigningAlg: types.CertAuthoritySpecV2_RSA_SHA2_256,
+		Roles: nil,
```
Comment (codingllama) [inline lib/services/watcher_test.go:869]:
```suggestion
```

File: tool/tctl/common/auth_command_test.go [hunk 8]
```diff
@@ -66,8 +66,7 @@ func TestAuthSignKubeconfig(t *testing.T) {
 			SSH: []*types.SSHKeyPair{{PublicKey: []byte("SSH CA cert")}},
 			TLS: []*types.TLSKeyPair{{Cert: cert}},
 		},
-		Roles:      nil,
-		SigningAlg: types.CertAuthoritySpecV2_RSA_SHA2_512,
+		Roles: nil,
```
Comment (codingllama) [inline tool/tctl/common/auth_command_test.go:69]:
```suggestion
```

**Review by codingllama — APPROVED** [review]
LGTM.

All suggestions are about removing redundant initializations. Feel free to push back/ignore as you see fit.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**